lib/Doctrine/Common/Collections/Collection.php

```diff
@@ -286,9 +286,9 @@ public function indexOf($element);
      * @param int      $offset The offset to start from.
      * @param int|null $length The maximum number of elements to return, or null for no limit.
      *
-     * @return array
+     * @return Collection
      *
-     * @psalm-return array<TKey,T>
+     * @psalm-return Collection<TKey, T>
```

Comment:
Should we add `@return static` here, and then change the type hints of implementors to `self` when dropping php 7.3?

Reply:
I am not sure. Can that be done in another issue/PR?

Reply:
Sure! It's a tiny detail, definitely not a blocker